Hide long data format via tooltip

timur987 opened the pull request:
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to SELFSERVE-613

## Proposed Changes

* Make the date format shorted, and
* Hide the long version under a tooltip.

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Run Calypso locally and opens Tools > Advertising.
* Ensure date format looks the same as in [Figma](https://www.figma.com/file/uvh2KtY65b7hGBHI9exbJ1/BlazePress?type=design&node-id=8572-76417&t=vSRZCWOBuy4WLzwg-0).

![date-tooltip](https://github.com/Automattic/wp-calypso/assets/115007291/b4749e82-78dc-4d02-a665-42b58ac216b6)

* (Optional) To limit unit tests to run only the new tests you can adjust [`package.json:116`](https://github.com/Automattic/wp-calypso/blob/trunk/package.json#L116) to become:
```json
"test-client": "TZ=UTC jest -c=test/client/jest.config.js -i client/my-sites/promote-post-i2/test/utils.test.tsx",
```

* Run unit tests to make sure the new test is working fine by running (if there's a way to run specified unit tests without changing `package.json` please let me know):
```sh
yarn run test-client
``` 

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [x] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

timur987 committed:
Hide long data format via tooltip

sbarbosa commented on client/my-sites/promote-post-i2/components/relative-time/index.tsx:
```diff
@@ -0,0 +1,70 @@
+	);
+}
+
+RelativeTime.propTypes = {
```
We are already using typescript for this component. I don't think we need propTypes here.

timur987 replied:
Oh, indeed. Thanks for caching this! Addressed in https://github.com/Automattic/wp-calypso/pull/78501/commits/939433ced1d3e9bcd117e25809263e51d87f68de

sbarbosa reviewed (approved):
LGTM 👍

Left a minor non-blocking comment.

kthai15 commented:
Mentioning this here as we had to do the same process in the widget. PR [here](https://github.tumblr.net/Tumblr/a8c-dsp/pull/214).

![9gg6ag.gif](https://github.com/Automattic/wp-calypso/assets/2396764/2f909885-d766-440a-a5ad-191443a4d434)

- In the widget, we have the tooltip opening on hover over the date, not on click
- The logic that displays the text in the table, where `dateDiff` is the difference in days between the post publish date and the current datetime, and `isWithinCurrentYear` checks if the publish date year is the same as the current year.

```
{dateDiff === 0 && translate("hours ago")}
{dateDiff === 1 && translate("1 day ago")}
{dateDiff > 1 && isWithinCurrentYear && moment(date).format("MMM DD")}
{dateDiff > 1 && !isWithinCurrentYear && moment(date).format("MMM DD, YYYY")}
```

We'll probably want to keep this consistent between Calypso and the widget, so perhaps we can touch base on this after the launch tomorrow 😃

timur987 committed:
Address PR comments

timur987 commented:
Hey @kthai15!

Thanks for the feedback.

> In the widget, we have the tooltip opening on hover over the date, not on click
Sure, I've changed it.

> The logic that displays the text in the table, where dateDiff is the difference in days between the post publish date and the current datetime, and isWithinCurrentYear checks if the publish date year is the same as the current year.

Thanks for the suggestion. Previously I used the previous implementation, but it's great to have the same experience as in the [new widget](https://github.tumblr.net/Tumblr/a8c-dsp/pull/214/files#diff-7f3a28a360a57de04acfa981da4a6ee9a5fefe4e08e2a1c69dfcdf357a0eb8c1R61).

I addressed all suggestions within https://github.com/Automattic/wp-calypso/pull/78501/commits/939433ced1d3e9bcd117e25809263e51d87f68de (they will be anyway squashed)

timur987 committed:
Merge remote-tracking branch 'origin/trunk' into update/blazepress-data-format

sbarbosa commented:
Nice, the changes look good 👍

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/8061877

Thank you @timur987 for including a screenshot in the description! This is really helpful for our translators.

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/8061877).
